File: Task/Determine-if-a-string-has-all-the-same-characters/Java/determine-if-a-string-has-all-the-same-characters-2.java
public static void main(String[] args) {
    String[] strings = {
        "", "   ", "2", "333", ".55", "tttTTT", "5", "4444 444k"
    };
    for (String string : strings)
        System.out.println(printCompare(string));
}

static String printCompare(String string) {
    String stringA = "'%s' %d".formatted(string, string.length());
    Pattern pattern = Pattern.compile("(.)\\1*");
    Matcher matcher = pattern.matcher(string);
    StringBuilder stringB = new StringBuilder();
    /* 'Matcher' works dynamically, so we'll have to denote a change */
    boolean difference = false;
    char character;
    String newline = System.lineSeparator();
    while (matcher.find()) {
        if (matcher.start() != 0) {
            character = matcher.group(1).charAt(0);
            stringB.append(newline);
            stringB.append("  Char '%s' (0x%x)".formatted(character, (int) character));
            stringB.append(" @ index %d".formatted(matcher.start()));
            difference = true;
        }
    }
    if (!difference)
        stringB.append(newline).append("  All characters are the same");
    return stringA + stringB;
}

File: Task/Determine-if-a-string-has-all-the-same-characters/Standard-ML/determine-if-a-string-has-all-the-same-characters-2.ml
- map chkstring [ "","   ","1","333",".55","tttTTT","4444 444k" ];
val it =
   [("", 0, allSame), ("   ", 3, allSame), ("1", 1, allSame),
    ("333", 3, allSame), (".55", 3, difference ("35", #"5", 2)),
    ("tttTTT", 6, difference ("54", #"T", 4)),
    ("4444 444k", 9, difference ("20", #" ", 5))]:
